# Calculate format=diff merge(app-arch/dump)==0.4.46&&pkg(dev-libs/openssl)>=1.1
diff -Naur dump-0.4b46/common/transformation_ssl.c dump-0.4b46-new/common/transformation_ssl.c
--- dump-0.4b46/common/transformation_ssl.c	2016-06-08 07:01:45.000000000 +0200
+++ dump-0.4b46-new/common/transformation_ssl.c	2019-01-02 11:26:15.945017945 +0100
@@ -215,7 +215,7 @@
 	/* to be exposed to any attacker anyway. */
 	*saltlen = 16;
 	if (xform->enc == 1) {
-		RAND_pseudo_bytes(salt, *saltlen);
+		RAND_bytes(salt, *saltlen);
 	}
 	memcpy(ivbuffer, salt, 16);
 
@@ -515,7 +515,7 @@
 		//EVP_CIPHER_CTX_rand_key(ctx, t->state.ssl.key);
 		//EVP_CIPHER_CTX_cleanup(ctx);
 		//EVP_CIPHER_CTX_free(ctx);
-		RAND_bytes(t->state.ssl.key, t->state.ssl.cipher->key_len);
+		    RAND_bytes(t->state.ssl.key, EVP_CIPHER_key_length(t->state.ssl.cipher));
 	} else {
 		// how do we get keys?
 	}
